### DISCUSSION TITLE
DevOps 소통 도구에 관한 생각

### DISCUSSION THREAD

**Discussion by sokojh** [heading1/WYLSBingsu#9] [Ideas]
# 서론
그 동안 많은 프로젝트를 진행한 것은 아니지만, 하나를 하더라도

좀 더 현업에 가깝게 진행을 하고 싶어, 여러가지 DevOps 도구들을 가지고 진행 한적이 있습니다. 

저랑 같이한 팀원분들은 잘 아실것이라고 생각합니다. 😏

주로 Slack, Jira, Notion 등  현업에서 쓰는 도구를 많이 써보자 라는 생각이었습니다.

# 문제
하지만... 이런 도구를 많이 쓰다보니, 

개발을 진행하며 어디에 집중을 해야하는지, 서로 피드백은 잘 되고 있는지 트랙킹이 전혀 되질 않았습니다. 

도구에 대한 숙련도가 적다고 그런거라고 생각할수도 있었지만, 

그렇게 어려운 숙련도를 요구하는 것이 아님에도

팀원 모두가 잘 사용하고 소통이 된다는 느낌을 받아보질 못했습니다.

즉, 도구가 많다보니 이리저리 도구를 전환하는 과정에서 오버헤드가 생겼다는 원인을 파악했습니다.

# 해결방안
그래서 이번엔 같은 실수를 반복하지 않으려고 한가지 도구에서 처리할 수 있으면 처리해보자라는 생각이 들었습니다. (github Action까지!)

한가지 도구에서 처리하게되면 기대되는 5가지 효과입니다.

|번호|효과|
|:--:|---|
|1|어디에 중요한 정보가 오는지 고민하지 않는다.|
|2|모든 인원이 이 도구의 트랙킹을 잘하고 있는지 고민하지 않는다.|
|3| 한가지 도구에서 모든 일처리를 하기 때문에 정보유실이 없다.|
|4|한가지 도구만 사용하기 때문에 도구의 숙련도 또한 증가된다.|
|5| 숙련도가 증가되면 생산성 또한 증가된다.|

[저와 같은 고민을 한 카카오 엔터프라이즈 팀 글](https://tech.kakao.com/2022/05/06/github-actions/)

**Comment by Circlewee** [comment]
이전 프로젝트에서 익숙하지않은 여러 도구들을 동시에 활용하려하니 이도저도 아닌 상태가 되는 것 같더라고요...
하나에 집중해서 모든 것을 표현하는 것도 좋은 것 같습니다.

**Reply by sokojh** [reply]
> 이전 프로젝트에서 익숙하지않은 여러 도구들을 동시에 활용하려하니 이도저도 아닌 상태가 되는 것 같더라고요... 하나에 집중해서 모든 것을 표현하는 것도 좋은 것 같습니다.

너무 공감됩니다. 저도 욕심이 많을때는 모든 도구를 모든 팀원이나 저나  다 잘 쓸수 있다고 생각했었는데, 

전혀 아니더라구요 ㅋㅋㅋ... 

책에서도 봤던건데, 인간의 뇌는 멀티태스킹이 안되는 것이 정상이라고 하네요. [관련 기사](https://www.sciencetimes.co.kr/news/%eb%a9%80%ed%8b%b0%ed%83%9c%ec%8a%a4%ed%82%b9-%ed%95%9c%eb%b2%88%ec%97%90-%eb%91%90-%ea%b0%80%ec%a7%80%ea%b0%80-%eb%87%8c%ec%9d%98-%ed%95%9c%ea%b3%84/)

이번엔 성공적으로 일하고 싶습니당! 😁
